inference-engine/tests/functional/shared_test_classes/src/base/ov_layer_test_utils.cpp

```diff
@@ -0,0 +1,317 @@
+        std::copy(buffer, buffer + inputSize, referenceInput.data());
+        refInputsTypes[i] = input.get_element_type();
+    }
+
+    std::vector<std::pair<element::Type, std::vector<std::uint8_t>>> expectedOutputs;
+    switch (refMode) {
+        case RefMode::INTERPRETER: {
+            expectedOutputs = ngraph::helpers::interpreterFunction(function, referenceInputs, refInputsTypes);
+            break;
+        }
+        case RefMode::CONSTANT_FOLDING: {
+            const auto &foldedFunc = ngraph::helpers::foldFunction(function, referenceInputs, refInputsTypes);
+            expectedOutputs = ngraph::helpers::getConstData(foldedFunc);
+            break;
+        }
+        case RefMode::IE: {
+            // reference inference on device with other options and nGraph function has to be implemented here
+            break;
+        }
+    }
+
+    return expectedOutputs;
+}
+
+std::vector<ov::runtime::Tensor> LayerTestsCommon::get_outputs() {
+    auto outputs = std::vector<ov::runtime::Tensor>{};
+    for (const auto& output : executableNetwork.get_results()) {
+        const auto& name = output->input_value(0).get_node()->get_friendly_name();
+        outputs.push_back(inferRequest.get_tensor(name));
+    }
+    return outputs;
+}
+
+void LayerTestsCommon::validate() {
+    auto expectedOutputs = calculate_refs();
+    const auto& actualOutputs = get_outputs();
+
+    if (expectedOutputs.empty()) {
+        return;
+    }
+
+    OPENVINO_ASSERT(actualOutputs.size() == expectedOutputs.size(),
+         "nGraph interpreter has ", expectedOutputs.size(), " outputs, while IE ", actualOutputs.size());
+
+    compare(expectedOutputs, actualOutputs);
+}
+
+std::string LayerTestsCommon::get_runtime_precision(const std::string& layerName) {
+    const auto function = executableNetwork.get_runtime_function();
+    for (const auto& op : function->get_ops()) {
+        const auto name = op->get_friendly_name();
+        if (name == layerName) {
+            const auto& rtInfo = op->get_rt_info();
+            const auto& it = rtInfo.find("runtimePrecision");
+
+            OPENVINO_ASSERT(it != rtInfo.end(), "Runtime precision is not found for node: ", name);
+
+            const auto rtPrecisionPtr = ngraph::as_type_ptr<ngraph::VariantWrapper<std::string>>(it->second);
+            return rtPrecisionPtr->get();
+        }
+    }
+    return "";
+}
+
+std::string LayerTestsCommon::get_runtime_precision_by_type(const std::string& layerType) {
+    const auto function = executableNetwork.get_runtime_function();
+
+    for (const auto& op : function->get_ops()) {
+        const auto& rtInfo = op->get_rt_info();
+        const auto& typeIt = rtInfo.find("layerType");
+
+        OPENVINO_ASSERT(typeIt != rtInfo.end(), "Layer is not found for type: ", layerType);
+
+        const auto type = ngraph::as_type_ptr<ngraph::VariantWrapper<std::string>>(typeIt->second)->get();
+        if (type == layerType) {
```

Comment:
And one question. What about three convolution in one function?

Reply:
If there is an issue in that copy-pasted code please clarify what should I do to fix it.